src/Orleans.Core/ApplicationParts/ApplicationPartManagerExtensions.cs

```diff
@@ -226,40 +164,67 @@ void LoadReferencedAssemblies(Assembly asm, HashSet<Assembly> includedAssemblies
             }
         }
 
-        /// <summary>
-        /// Adds all assemblies referencing Orleans found in the application's <see cref="DependencyContext"/>.
-        /// </summary>
-        /// <param name="manager">The builder.</param>
-        /// <returns>The builder with the additionally included assemblies.</returns>
-        public static IApplicationPartManagerWithAssemblies AddFromDependencyContext(this IApplicationPartManager manager)
+        public static IApplicationPartManagerWithAssemblies AddFromAppDomainWithReferences(this IApplicationPartManager manager)
         {
-            return manager.AddFromDependencyContext(Assembly.GetCallingAssembly())
-                .AddFromDependencyContext(Assembly.GetEntryAssembly())
-                .AddFromDependencyContext(Assembly.GetExecutingAssembly());
+            var assemblies = new List<Assembly>();
+            foreach (var asm in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                manager = manager.AddApplicationPart(asm).WithReferences();
+                assemblies.Add(asm);
+            }
+
+            return new ApplicationPartManagerWithAssemblies(manager, assemblies);
+        }
+
+        public static IApplicationPartManagerWithAssemblies AddFromAssemblyLoadContext(this IApplicationPartManager manager, Assembly assembly = null)
+        {
+#if NETCOREAPP
+            assembly ??= typeof(ApplicationPartManagerExtensions).Assembly;
+            var assemblies = new HashSet<Assembly>();
+            var context = AssemblyLoadContext.GetLoadContext(assembly);
+            foreach (var asm in context.Assemblies)
+            {
+                manager = manager.AddApplicationPart(asm);
+                assemblies.Add(asm);
+            }
+
+            return new ApplicationPartManagerWithAssemblies(manager, assemblies);
+#else
+            throw new PlatformNotSupportedException("AssemblyLoadContext is not supported on this platform");
```

Comment:
Why not hide this method completely? Or put an `Obsolete` attribute if not using netcore, to get a clear error during compilation?

Reply:
You're right. I figure we will delete this whole thing very soon, so it's temporary either way